components/org.wso2.carbon.identity.login.resolver/org.wso2.carbon.identity.login.resolver.service/src/main/java/org/wso2/carbon/identity/login/resolver/service/LoginResolverServiceImpl.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright (c) 2023, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.login.resolver.service;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.event.IdentityEventException;
+import org.wso2.carbon.identity.login.resolver.mgt.LoginResolver;
+import org.wso2.carbon.identity.login.resolver.mgt.LoginResolverService;
+import org.wso2.carbon.identity.login.resolver.mgt.ResolvedUserResult;
+import org.wso2.carbon.identity.login.resolver.service.constants.LoginResolverServiceConstants;
+import org.wso2.carbon.identity.login.resolver.service.util.LoginResolverServiceUtil;
+import org.wso2.carbon.user.core.common.AuthenticationResult;
+
+import java.util.List;
+
+/**
+ * Provides the services which are needed for the login resolver functionalities.
+ */
+public class LoginResolverServiceImpl implements LoginResolverService {
+
+    private static final Log log = LogFactory.getLog(LoginResolverServiceImpl.class);
+
+    /**
+     * This method is used to determine whether the login resolver feature is enabled or not.
+     *
+     * @param tenantDomain The tenant domain of the user.
+     * @return True if the login resolver feature is enabled for given tenant, false if else.
+     */
+    @Override
+    public boolean isEnabled(String tenantDomain) {
+
+        if (StringUtils.isNotBlank(tenantDomain)) {
+            try {
+                return Boolean.parseBoolean(LoginResolverServiceUtil.getConnectorConfig(LoginResolverServiceConstants
+                        .LOGIN_RESOLVER_PROPERTY, tenantDomain));
+            } catch (IdentityEventException e) {
+                log.error("An error occurred while retrieving login resolver property.", e);
```

Comment:
Shouldn't we break the flow when we receive this error ? are we going to continue the flow ? WDYT ?

Reply:
In this case, the flow will not get triggered at all because the referring source code will always check whether the login resolver is enabled and will only execute the flow if it is enabled. So when there is an error, it will return false while printing the error message. As example, please refer to the code section at [1].

[1] - https://github.com/deshankoswatte/identity-governance/blob/b0aa7a681658f52dfeb520960fdb16f70df288b0/components/org.wso2.carbon.identity.api.user.recovery/src/main/java/org/wso2/carbon/identity/recovery/endpoint/impl/RecoverPasswordApiServiceImpl.java#L52.